mail/views_test.py

```diff
@@ -229,13 +231,62 @@ def test_automatic_email_fail(self):
             "email_subject": self.request_data['email_subject'],
             "email_body": self.request_data['email_body'],
             "sender_name": full_name(self.staff),
+            "staff_user": self.staff,
         }
 
         assert sorted(SentAutomaticEmail.objects.filter(
             user__email__in=self.email_results, automatic_email=self.automatic_email,
         ).values_list('user__email', flat=True)) == sorted(success_emails)
 
 
+class AutomaticEmailViewTests(APITestCase, MockedESTestCase):
+    """
+    AutomaticEmailViewTests
+    """
+
+    def setUp(self):
+        self.staff_user = UserFactory.create()
+        self.program = ProgramFactory.create()
+        Role.objects.create(
+            user=self.staff_user,
+            program=self.program,
+            role=Staff.ROLE_ID,
+        )
+        self.url = reverse('automatic_email_api')
```

Comment:
everything you set up until here can be moved to a `setUpTestData` classmethod. The benefit is that it will run only once for all the tests in this class